plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java

```diff
@@ -307,12 +309,21 @@ private Schema toIcebergSchema(List<Column> columns)
         return new Schema(icebergSchema.asStructType().fields());
     }
 
-    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId)
+    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId, HiveStorageFormat storageFormat)
     {
         if (type instanceof ArrayType || type instanceof MapType || type instanceof RowType) {
             // TODO https://github.com/trinodb/trino/issues/17583 Add support for these complex types
             throw new TrinoException(NOT_SUPPORTED, "Migrating %s type is not supported".formatted(type));
         }
+
+        // TODO: Add support for structural types with timestamp(3)
+        if (type.equals(TimestampType.TIMESTAMP_MILLIS)) {
+            // TODO Adding support for Avro is currently suspended as there are parallel works on switching to Avro native reader. https://github.com/trinodb/trino/pull/17221
+            return switch (storageFormat) {
+                case ORC, PARQUET -> Types.TimestampType.withoutZone();
+                default -> throw new TrinoException(NOT_SUPPORTED, "Migrating timestamp(3) type with %s format is not supported. Use \"timestamp(6)\" instead.".formatted(storageFormat));
```

Comment:
1. don't use `defautl`. use `AVRO` here
2. why not reurn `Types.TimestampType.withoutZone()` always? i.e. what would happen if we do not filter out Avro here?

Reply:
Please see https://github.com/trinodb/trino/pull/17391#discussion_r1206571775